kubernetes/base/leaderelection/leaderelection.py

```diff
@@ -103,6 +107,15 @@ def renew_loop(self):
 
             # failed to renew, return
             return
+    
+    def release(self):
+        lock_status, old_election_record = self.election_config.lock.get(self.election_config.lock.name,
+                                                                        self.election_config.lock.namespace)
+        if lock_status:
+            logging.info("release lock: {}".format(old_election_record.holder_identity))
+            old_election_record.lease_duration = 1
+            old_election_record.holder_identity = None
```

Comment:
the lock may have been successfully acquired by another client. Doing so may force that client to lose the leadership. Why do we want to do this?

Reply:
this happens when lock has been acquired and exception occurs that lock cannot be normally release, if not release the lock here, the next leader election must wait for timeout

Reply:
i fix it just like client-go did: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go#L294